lumicks/pylake/channel.py

```diff
@@ -504,6 +504,172 @@ def plot(self, start=None, **kwargs):
         plt.ylabel(self.labels.get("y", "y"))
         plt.title(self.labels.get("title", "title"))
 
+    @staticmethod
+    def _annotate(start_time, annotation, annotation_direction, stop_time, color):
+        """Annotate a region or time point with some text
+
+        Parameters
+        ----------
+        start_time : float
+            Start time in seconds
+        annotation : str | None
+            String to annotate with
+        annotation_direction : "horizontal" | "vertical" | None
+            Which direction to show the text when annotating.
+        stop_time : float
+            Stop time in seconds
+        color : matplotlib.color | None
+            Matplotlib color
+        """
+
+        import matplotlib.pyplot as plt
+
+        ax = plt.gca()
+        annotation_args = {
+            "va": "top",
+            "y": 0.98,
+            "s": annotation,
+            "transform": ax.get_xaxis_transform(),
+            "color": color,
+        }
+        if not annotation_direction:
+            annotation_direction = "horizontal" if stop_time else "vertical"
+
+        match annotation_direction:
+            case "horizontal":
+                ax.text(
+                    ((start_time + stop_time) / 2 if stop_time else start_time),
+                    ha="center",
+                    rotation=0,
+                    **annotation_args,
+                )
+            case "vertical":
+                if stop_time:
+                    # Prefer using the stop time, as that puts the text _inside_ the shaded area
+                    ax.text(stop_time, ha="right", rotation=90, **annotation_args)
+                else:
+                    ax.text(start_time, ha="right", rotation=90, **annotation_args)
+            case _:
+                raise RuntimeError(
+                    'Invalid value passed for annotation_direction. Expected "horizontal" or '
+                    f'"vertical", got {annotation_direction}.'
+                )
+
+    def highlight_time_range(
+        self,
+        item,
+        annotation=None,
+        annotation_direction=None,
+        color=None,
+        label=None,
+        start=None,
+        alpha=0.5,
+        **kwargs,
+    ):
+        """Highlight some points or regions in a plot.
+
+        Parameters
+        ----------
+        item : Marker | tuple(int, int) | int | str
+            An item with a start and stop property such as a `Marker` or `CalibrationItem`, a
+            tuple of timestamps, a single timestamp or a string in time format (e.g. "5s" for
+            5 seconds).
+        annotation : str | None
+            String to annotate with
+        annotation_direction : "horizontal" | "vertical" | None
+            Which direction to show the text when annotating.
+        color : matplotlib.color | None
+            Matplotlib color
```

Comment:
this is a bit vague as it doesn't say what the color is applied to. see my comment below too

Reply:
Yeah, that's a fair point. It's gone now. Ideally, I would still like there to be a "default" option that just gets applied to all (the convenience option). So I would propose forwarding the color to `annotation_kwargs` if it does not specify colors itself.